rest-api/dao/questionnaire_response_dao.py

```diff
@@ -106,8 +142,16 @@ def insert_with_session(self, session, questionnaire_response):
     resource_json['id'] = str(questionnaire_response.questionnaireResponseId)
     questionnaire_response.resource = json.dumps(resource_json)
 
+    # Gather the question ids and records that match the questions in the response
     question_ids = [answer.questionId for answer in questionnaire_response.answers]
     questions = QuestionnaireQuestionDao().get_all_with_session(session, question_ids)
+
+    # DA-623: raise error when response link ids do not match our question link ids.
+    # Gather the valid link ids for this question
+    link_ids = [question.linkId for question in questions]
+    # look through the response and verify each link id is valid for each question.
+    self._validate_link_ids_from_resource_json_group(resource_json, link_ids)
```

Comment:
How confident are we that this won't immediately start firing for all incoming questionnaire's? Or would incoming data in this format currently cause major data issues?

I'm not sure off-hand whether we might expect to find other values in here. If there's a risk here, consider instead pushing this initially with just a try/catch/log to monitor.

Reply:
I ran tests with recent questionnaires/response records in stable and all passed this validation check. This certainly restricts behavior that was taken advantage of in the unit tests. We are checking with PTSC to get their feedback before merging.  It may be the final decision to log for a period of time before actually enforcing the validation.

Reply:
cool 👍